packages/indexer/tests/db_test.py

```diff
@@ -24,77 +35,133 @@ async def mock_pool():
+
+
+@pytest.fixture
+async def mock_db(mock_pool: asyncpg.Pool):
+    db = Database(pool=mock_pool, schema=os.getenv("INDEXER_SCHEMA", "public"))
+    await db.pool.execute(f"CREATE SCHEMA IF NOT EXISTS {db.schema}")
+    await drop_tables(db)
+    return db
+
+
+@pytest.fixture
+async def mock_notification_handler(mock_db: Database):
+    dbNotifHandler = DbNotificationHandler(mock_db)
+    return dbNotifHandler
+
+
 @pytest.fixture
 def raw_blocks():
-    with open("tests/test_data/raw_blocks.json", "r") as f:
-        return json.load(f)
+    df = pd.read_csv("tests/test_data/raw_blocks.csv")
+    df["block"] = df["block"].apply(json.loads)
+    return df
+
+
+@pytest.fixture
+def raw_txs():
+    df = pd.read_csv("tests/test_data/raw_txs.csv")
+    df["txs"] = df["txs"].apply(lambda x: json.loads(x) if not pd.isna(x) else [])
+    return df
 
 
 @pytest.mark.asyncio
-async def test_create_drop_tables(mock_pool: asyncpg.pool):
-    async def check_tables(pool, table_names, schema) -> int:
-        async with pool.acquire() as conn:
+async def test_create_drop_tables(mock_db: Database):
+    async def check_tables(mock_db: Database, table_names) -> int:
+        async with mock_db.pool.acquire() as conn:
             results = await conn.fetch(
                 f"""
                 SELECT table_name FROM information_schema.tables 
-                WHERE table_schema = $1
+                WHERE table_schema = '{mock_db.schema}'
                 AND table_name in {table_names}
-                """,
-                schema,
+                """
             )
         return len(results)
 
-    await create_tables(mock_pool)
+    await create_tables(mock_db)
 
     table_names = ("raw_blocks", "raw_txs", "blocks", "txs", "logs", "log_columns")
 
-    assert await check_tables(mock_pool, table_names, "public") == len(table_names)
+    assert await check_tables(mock_db, table_names) == len(table_names)
 
-    assert await get_table_cols(mock_pool, "blocks") == [
+    assert await get_table_cols(mock_db, "blocks") == [
         "height",
         "chain_id",
         "time",
         "block_hash",
         "proposer_address",
     ]
 
-    await drop_tables(mock_pool)
+    await drop_tables(mock_db)
 
-    assert await check_tables(mock_pool, table_names, "public") == 0
+    assert await check_tables(mock_db, table_names) == 0
 
 
 @pytest.mark.asyncio
 async def test_upsert_raw_blocks(
-    mock_pool,
-    raw_blocks,
+    mock_db: Database,
+    mock_chain: CosmosChain,
+    mock_client: aiohttp.ClientSession,
+    mock_semaphore: asyncio.Semaphore,
+    raw_blocks: pd.DataFrame,
+    mocker,
 ):
-    await create_tables(mock_pool)
-    for block in raw_blocks:
-        await upsert_raw_blocks(mock_pool, block)
+    await create_tables(mock_db)
+    for block in raw_blocks["block"][: len(raw_blocks) // 2]:
+        await upsert_raw_blocks(mock_db, block)
+
+    # valid process block
+    for height, _chain_id, block, _parsed_at in raw_blocks[
```

Comment:
Split testing data into two sections to test two different implementations

Then test incorrect data

Reply:
Good